### PR TITLE
[fix][dataloader] use file descripor instead of file system

### PR DESCRIPTION
**Pull request by xysheng-baidu**
### PR Category

User Experience

### PR Types

Bug fixes

### Description

PCard-81023
use file descripor instead of file system in dataloader.

### CODE REVIEW

File: paddle/common/flags.cc [hunk 1]
```diff
@@ -1345,6 +1345,19 @@ PHI_DEFINE_EXPORTED_bool(use_shm_cache,
...
+/**
+ * mmap_allocator related FLAG
+ * Name: use_file_descripor
+ * Since Version: 2.6.2
+ * Value Range: bool, default=true
+ * Example:
+ * Note: . If True, mmap_allocator will use file descripor to open shared memory
+ * operation.
+ */
+PHI_DEFINE_EXPORTED_bool(use_file_descripor,
```
Comment (sneaxiy) [inline paddle/common/flags.cc]: Change it be `dataloader_use_file_descriptor`.

File: python/paddle/incubate/multiprocessing/reductions.py [hunk 2]
```diff
@@ -161,11 +169,31 @@ def _reduce_lodtensor(lodtensor):
             if dim == 0:
                 # Empty tensors have nothing be mapped.
                 return (_rebuild_lodtensor_empty, (type(lodtensor),))
+        inmeta = False
+        if os.environ.get('FLAGS_use_file_descripor', True) in [
```
Comment (sneaxiy) [inline python/paddle/incubate/multiprocessing/reductions.py]: Change it be `core.globals()["FLAGS_use_file_descripor"]`. The `core` is `paddle.base.core`.

File: python/paddle/incubate/multiprocessing/reductions.py [hunk 3]
```diff
@@ -161,11 +169,31 @@ def _reduce_lodtensor(lodtensor):
...
 
         # Default use share filename strategy
-        metadata = (
-            lodtensor._share_filename()
-        )  # ipc_name, size, type_idx, dims, lod
+        metadata = lodtensor._share_filename(
+            inmeta
+        )  # ipc_name, fd, size, type_idx, dims, lod
+
+        if os.environ.get('FLAGS_use_file_descripor', True) in [
```
Comment (sneaxiy) [inline python/paddle/incubate/multiprocessing/reductions.py]: Change it be `core.globals()["FLAGS_use_file_descripor"]`. The `core` is `paddle.base.core`.

**Review by sneaxiy — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]
LGTM
单测超时时间设置

**Review by lanxianghit — APPROVED** [review]
LGTM for Flags

**Pull request merged**